core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixTextToSpeech.java

```diff
@@ -133,13 +135,21 @@ public void readAloud(WebView webView) {
       } else {
         tts.setLanguage(locale);
 
+        if (getFeatures(tts,locale).contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED)){
+          ContextExtensionsKt.toast(this.context,R.string.tts_lang_not_supported,Toast.LENGTH_LONG);
+          return;
+        }
+
         if (requestAudioFocus()) {
           loadURL(webView);
         }
       }
     }
   }
-
+  @SuppressLint("NewApi")
+  private Set<String> getFeatures(TextToSpeech tts, Locale locale){
+    return ((VERSION.SDK_INT < VERSION_CODES.LOLLIPOP)?tts.getFeatures(locale):tts.getVoice().getFeatures());
```

Comment:
press `ctrl+alt+l` to auoformat this, there should be a space on either side of an operator.
Also delete these unnecessary brackets around `VERSION.SDK_INT < VERSION_CODES.LOLLIPOP` and the brackets around the ternanry